atl-checker/src/algorithms/game_strategy/partial.rs

```diff
@@ -0,0 +1,275 @@
+                    // We skip the first edge with the `until` formula target. We know that it is false.
+                    #[cfg(debug_assertions)]
+                    {
+                        if let AnnotatedEdge::Hyper(e) = edges.get(0).unwrap() {
+                            let t = e.targets.get(0).unwrap();
+                            let a = assignments.get(&t.0);
+                            match a {
+                                Some(VertexAssignment::False) => {}
+                                _ => panic!(
+                                    "First edge of despite was not false contrary to assumption"
+                                ),
+                            }
+                        }
+                    }
+                    for edge in edges.iter().skip(1) {
+                        if let AnnotatedEdge::Hyper(edge) = edge {
+                            // If the `pre` formula is also false, then we don't need to check further
+                            let pre = edge.targets.get(0).unwrap();
+                            if let Some(VertexAssignment::False) = assignments.get(&pre.0) {
+                                return;
+                            }
+
+                            // We expect to find infinite computations that never hits the `until` formula.
+                            // This means there will be a cycle of undecided configurations.
+                            // Find the first target assigned undecided.
+                            for target in edge.targets.iter().skip(1) {
+                                if matches!(
+                                    assignments.get(&target.0),
+                                    Some(VertexAssignment::Undecided)
+                                ) {
+                                    // Target is annotated with the move we need
+                                    move_to_pick.insert(*state, target.1.as_ref().unwrap().clone());
+                                    compute_partial_strategy_rec(
+                                        graph,
+                                        &target.0,
+                                        assignments,
+                                        move_to_pick,
+                                    );
+                                    return;
+                                }
+                            }
+                        }
+                    }
+                }
+                _ => panic!("The formula does not require a strategy"),
+            }
+        }
+        AtlVertex::Partial { .. } => {
+            // Partial vertices are only used by despite formulae.
+            // The full vertices handles finding the moves so we just need to visit all
+            // targets of all the edges to proceed with the computation in the successor states.
+            let edges = graph.annotated_succ(vertex);
+            for edge in edges {
```

Comment:
Same as before, no debug used here so it could be
```suggestion
            for edge in graph.annotated_succ(vertex) {
```